tests/test_category_checker.py

```diff
@@ -1,137 +1,58 @@
-                "categories": [],
-            }
-        ]
-        fc = generate_data_as_json_file(data=data)
-        checker = check_config(file_config=fc, info=False)
-        self.assertFalse(checker)
-
-    def test_5C_duplicate_with_5CD(self):
-        data = [
-            {
-                "name": "Water",
-                "form": 1,
                 "categories": [
                     {
-                        "name": "Category-1A",
-                        "and": [
-                            {"question": 1, "options": ["A"]},
+                        "questions": [
+                            {
+                                "id": 1,
+                                "options": ["A", "B"],
+                                "other": [{"questions": []}],
+                                "else": {},
+                            }
                         ],
-                    },
+                    }
                 ],
-            },
-            {
-                "name": "Sanitation",
-                "form": 1,
-                "categories": [
-                    {
-                        "name": "Category-5C",
-                        "and": [
-                            {"question": 5, "options": ["C"]},
-                        ],
-                    },
-                    {
-                        "name": "Category-5CD",
-                        "and": [
-                            {"question": 5, "options": ["C", "D"]},
-                        ],
-                    },
-                ],
-            },
+            }
         ]
         fc = generate_data_as_json_file(data=data)
-        checker = check_config(file_config=fc, info=False)
-        self.assertEqual(checker, 1)
+        errors, questions = check_config(file_config=fc, info=False)
+        self.assertEqual(len(errors), 3)
 
-    def test_1A_duplicate_with_1AB_and_1A_with_1AB2CD(self):
+    def test_duplicate_question_11CD_with_11FG(self):
+        # Different form ids have the same question id.
         data = [
             {
-                "name": "Water",
                 "form": 1,
                 "categories": [
                     {
-                        "name": "Category-1A",
-                        "and": [
-                            {"question": 1, "options": ["A"]},
-                        ],
-                    },
-                    {
-                        "name": "Category-1AB",
-                        "and": [{"question": 1, "options": ["A", "B"]}],
-                    },
-                    {
-                        "name": "Category-1AB2CD",
-                        "and": [
-                            {"question": 1, "options": ["A", "B"]},
-                            {"question": 2, "options": ["C", "D"]},
-                        ],
-                    },
+                        "name": "Category-11CD",
+                        "questions": [{"id": 11, "options": ["C", "D"]}],
```

Comment:
This test is wrong, how different form can have same question id?. The error should check if this question id is part of the form before checking the duplication. Check duplication should be in the same categories list.